### PR TITLE
[Merged by Bors] - feat: The multinomial theorem in terms of `piAntidiag`

### PR DESCRIPTION
**Pull request by YaelDillies**
Prove that `(∑ i in s, f i) ^ n = ∑ k in piAntidiag s n, multinomial s k * ∏ i in s, f i ^ k i`. We already have a version of this for `Finset.sym`, so I am also adding machinery to deduce one from the other. I however haven't managed to do that.

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finset/NoncommProd.lean [hunk 1]
```diff
@@ -162,15 +162,15 @@ lemma noncommProd_induction (s : Multiset α) (comm)
...
 @[to_additive]
-protected theorem noncommProd_map_aux [MonoidHomClass F α β] (s : Multiset α)
+protected theorem map_noncommProd_aux [MonoidHomClass F α β] (s : Multiset α)
     (comm : { x | x ∈ s }.Pairwise Commute) (f : F) : { x | x ∈ s.map f }.Pairwise Commute := by
   simp only [Multiset.mem_map]
   rintro _ ⟨x, hx, rfl⟩ _ ⟨y, hy, rfl⟩ _
   exact (comm.of_refl hx hy).map f
 
 @[to_additive]
-theorem noncommProd_map [MonoidHomClass F α β] (s : Multiset α) (comm) (f : F) :
```
Comment (eric-wieser) [inline Mathlib/Data/Finset/NoncommProd.lean:172]: Please fill out the renames section of the PR description; and preferably do this rename in a standalone PR.

Comment (YaelDillies) [reply Mathlib/Data/Finset/NoncommProd.lean]: Split out to #15048

File: Mathlib/Data/Nat/Choose/Multinomial.lean [hunk 2]
```diff
@@ -196,16 +197,58 @@ theorem multinomial_zero [DecidableEq α] : multinomial (0 : Multiset α) = 1 :=
...
+    (∑ i in s, f i) ^ n = ∑ k in piAntidiag s n, multinomial s k *
+      s.noncommProd (fun i ↦ f i ^ k i) (hc.mono' fun i j h ↦ h.pow_pow ..) := by
+  classical
+  induction' s using Finset.cons_induction with a s has ih generalizing n
+  · cases n <;> simp
+  rw [Finset.sum_cons, piAntidiag_cons, sum_disjiUnion]
+  simp only [sum_map, Function.Embedding.coeFn_mk, Pi.add_apply, multinomial_cons,
+    Pi.add_apply, eq_self_iff_true, if_true, Nat.cast_mul, noncommProd_cons, eq_self_iff_true,
+    if_true, sum_add_distrib, sum_ite_eq', has, if_false, add_zero,
+      addLeftEmbedding_eq_addRightEmbedding, addRightEmbedding_apply]
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Nat/Choose/Multinomial.lean:222]:
```suggestion
    addLeftEmbedding_eq_addRightEmbedding, addRightEmbedding_apply]
```

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer delegate

**Comment by YaelDillies** [comment]
I guess I should now be doing

maintainer merge

?

**Pull request closed without merging**